packages/dds/test-dds-utils/src/ddsSnapshotHarness.ts

```diff
@@ -0,0 +1,88 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { strict as assert } from "node:assert";
+import { existsSync, readFileSync, writeFileSync } from "node:fs";
+import path from "node:path";
+
+// Simple filter to avoid tests with a name that would accidentally be parsed as directory traversal or other confusing things.
+const nameCheck = new RegExp(/^[^"/\\]+$/);
+
+let currentTestName: string | undefined;
+let currentTestFile: string | undefined;
+
+/**
+ * @internal
+ */
+export interface ISnapshotSuite {
+	/**
+	 * A utility function for setting up the current snapshot directory for tests.
+	 */
+	useSnapshotDirectory: () => void;
+
+	/**
+	 * Takes a snapshot of data and writes it to a file.
+	 * @param data - The data to take a snapshot of.
+	 * @param writeCompatible - A flag indicating whether the snapshot should be checked for consistency
+	 * with old-format data. Defaults to true.
+	 */
+	takeSnapshot: (data: string, writeCompatible?: boolean) => string;
+}
+
+function takeSnapshot(data: string, writeCompatible: boolean = true): string {
+	assert(
+		currentTestName !== undefined,
+		"use `useSnapshotDirectory` to configure the tests containing describe block to take snapshots",
+	);
+	assert(currentTestFile !== undefined);
+
+	// Ensure test name doesn't accidentally navigate up directories or things like that.
+	// Done here instead of in beforeEach so errors surface better.
+	if (nameCheck.test(currentTestName) === false) {
+		assert.fail(`Expected test name to pass sanitization: "${currentTestName}"`);
+	}
+
+	if (!existsSync(currentTestFile)) {
+		writeFileSync(currentTestFile, data);
+	}
+	const pastData = readFileSync(currentTestFile, "utf8");
+	if (writeCompatible) {
+		assert.equal(data, pastData, `snapshots are inconsistent on test "${currentTestName}"`);
+	}
+	return data;
+}
+
+/**
+ * Creates a suite of functions for managing snapshots in tests.
+ * @param snapshotsFolder - The folder where snapshots will be stored.
+ * @returns An object containing functions for managing snapshots.
+ * @internal
+ */
+export function createSnapshotSuite(snapshotFolderPath: string): ISnapshotSuite {
+	/**
```

Comment:
It looks like this implementation lost some of the niceties that `snapshotTools.ts` had. Specifically:

- support for regenerating snapshots rather than asserting
- some sanity checks on using subdirectories, implemented here:

```
	// Basic sanity check to avoid bugs like accidentally recursively deleting everything under `/` if something went wrong (like dirPath navigated up directories a lot).
	assert(normalizedDir.startsWith(snapshotsFolder));
```

We should keep those features around.